src/adapters/MorphoMarketV1Adapter.sol

```diff
@@ -74,6 +75,8 @@ contract MorphoMarketV1Adapter is IMorphoMarketV1Adapter {
         require(msg.sender == parentVault, NotAuthorized());
         require(marketParams.loanToken == asset, LoanAssetMismatch());
 
+        if (shares[marketId] == 0 && assets > 0) allMarketParams.push(marketParams);
```

Comment:
If you allocate 1 asset that gets rounded to zero, then to remove the market from the `allMarketParams` array you will need to allocate more before you can withdraw. Otherwise when you try to deallocate 1 asset (to remove the market from the array) the withdraw will revert.